### PR TITLE
Feature/private only registry support

### PR DESCRIPTION
**Pull request by agduncan94**
Adds support for private registries and custom docker path registries.

- [x] Check that you pass the basic compilation and unit tests by running `grunt`

### CODE REVIEW

File: travisci/db_dump.sql [hunk 1]
```diff
@@ -1350,11 +1350,11 @@ SELECT pg_catalog.setval('token_id_seq', 4, true);
...
-INSERT INTO tool VALUES (4, NULL, NULL, NULL, NULL, 'git@github.com:DockstoreTestUser2/quayandgithubwdl.git', true, NULL, '2016-11-28 15:00:43.873', '/Dockstore.cwl', '/Dockerfile', '/Dockstore.wdl', '2016-03-15 15:36:22', 'AUTO_DETECT_QUAY_TAGS_AUTOMATED_BUILDS', 'quayandgithubwdl', 'dockstoretestuser2', 'quay.io/dockstoretestuser2/quayandgithubwdl', 'QUAY_IO', '', '', false);
-INSERT INTO tool VALUES (3, NULL, NULL, NULL, NULL, 'git@github.com:DockstoreTestUser2/quayandgithubalternate.git', false, NULL, '2016-11-28 15:02:48.557', '/Dockstore.cwl', '/testDir/Dockerfile', '/Dockstore.wdl', '2016-03-15 15:35:57', 'AUTO_DETECT_QUAY_TAGS_AUTOMATED_BUILDS', 'quayandgithubalternate', 'dockstoretestuser2', 'quay.io/dockstoretestuser2/quayandgithubalternate', 'QUAY_IO', '', '', false);
+INSERT INTO tool VALUES (1, 'Dockstore Test User', NULL, 'Whalesay deep quotes', NULL, 'git@github.com:A/a.git', false, NULL, '2016-11-28 15:00:43.873', '/Dockstore.cwl', '/Dockerfile', '/Dockstore.wdl', '2016-02-16 17:04:59', 'AUTO_DETECT_QUAY_TAGS_AUTOMATED_BUILDS', 'a', 'A', 'quay.io/A/a', 'QUAY_IO', '', '', false);
```
Comment (denis-yuen) [inline travisci/db_dump.sql]: Let's get these names changed

Comment (agduncan94) [inline travisci/db_dump.sql]: I've named these to preserve ordering, would something like A_namespace/a_name be better?

Comment (denis-yuen) [inline travisci/db_dump.sql]: Oh, not that part. I just meant to change 'Dockstore Test User'

**Comment by agduncan94** [comment]
The tests are all passing locally but not on travis.

**Pull request merged**